dask_sql/physical/rex/core/call.py

```diff
@@ -573,6 +573,25 @@ def overlay(self, s, replace, start, length=None):
         return s
 
 
+class CoalesceOperation(Operation):
+    def __init__(self):
+        super().__init__(self.coalesce)
+
+    def coalesce(self, *operands):
+        result = None
+        for operand in operands:
+            if is_frame(operand):
+                # Check if frame evaluates to nan or NA
+                if len(operand) == 1 and not operand.isnull().all().compute():
```

Comment:
In what case could we have an operand where the len is 1 but it's also a interpreted as a frame?

Reply:
This comes up when we pass through aggregations as input; if we attempt something like `coalesce(sum(b), 3)`, `sum(b)` will be passed through here as a len 1 series representing a "scalar", which we must somehow distinguish from a series that would otherwise represent a standard column.

There's some additional context in https://github.com/dask-contrib/dask-sql/pull/823#discussion_r1030672675, which I ended up resolving with https://github.com/dask-contrib/dask-sql/commit/2000932cd1e4adf5f0f57bb7b641443f0a7f5a7d

Reply:
Makes sense. It isn't ideal but I don't think there's a better way to accomplish this today